samples/extensions/logic_op_dynamic_state/logic_op_dynamic_state.cpp

```diff
@@ -0,0 +1,726 @@
+	        descriptor_sets.background,
+	        VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER,
+	        1,
+	        &background_image_descriptor)};
+
+	vkUpdateDescriptorSets(get_device().get_handle(), static_cast<uint32_t>(write_descriptor_sets.size()),
+	                       write_descriptor_sets.data(), 0, VK_NULL_HANDLE);
+}
+
+/**
+ * 	@fn void LogicOpDynamicState::model_data_creation()
+ * 	@brief Creating cube model
+ */
+void LogicOpDynamicState::model_data_creation()
+{
+	constexpr uint32_t vertex_count = 8;
+
+	std::array<glm::vec3, vertex_count> vertices_pos;
+	std::array<glm::vec3, vertex_count> vertices_norm;
+
+	vertices_pos[0] = {0.0f, 0.0f, 0.0f};
+	vertices_pos[1] = {1.0f, 0.0f, 0.0f};
+	vertices_pos[2] = {1.0f, 1.0f, 0.0f};
+	vertices_pos[3] = {0.0f, 1.0f, 0.0f};
+	vertices_pos[4] = {0.0f, 0.0f, 1.0f};
+	vertices_pos[5] = {1.0f, 0.0f, 1.0f};
+	vertices_pos[6] = {1.0f, 1.0f, 1.0f};
+	vertices_pos[7] = {0.0f, 1.0f, 1.0f};
+
+	/* Normalized normal vectors for each face of cube */
+	glm::vec3 Xp = {1.0, 0.0, 0.0};
+	glm::vec3 Xm = {-1.0, 0.0, 0.0};
+	glm::vec3 Yp = {0.0, 1.0, 0.0};
+	glm::vec3 Ym = {0.0, -1.0, 0.0};
+	glm::vec3 Zp = {0.0, 0.0, 1.0};
+	glm::vec3 Zm = {0.0, 0.0, -1.0};
+
+	/* Normalized normal vectors for each vertex (created by sum of corresponding faces) */
+	vertices_norm[0] = glm::normalize(Xm + Ym + Zm);
+	vertices_norm[1] = glm::normalize(Xp + Ym + Zm);
+	vertices_norm[2] = glm::normalize(Xp + Yp + Zm);
+	vertices_norm[3] = glm::normalize(Xm + Yp + Zm);
+	vertices_norm[4] = glm::normalize(Xm + Ym + Zp);
+	vertices_norm[5] = glm::normalize(Xp + Ym + Zp);
+	vertices_norm[6] = glm::normalize(Xp + Yp + Zp);
+	vertices_norm[7] = glm::normalize(Xm + Yp + Zp);
+
+	/* Scaling and position transform */
+	for (uint8_t i = 0; i < vertex_count; ++i)
+	{
+		vertices_pos[i] *= glm::vec3(8.0f, 8.0f, 8.0f);
+		vertices_pos[i] += glm::vec3(0.0f, 1.0f, 5.0f);
+	}
+
+	constexpr uint32_t index_count        = 29;
+	uint32_t           vertex_buffer_size = vertex_count * sizeof(glm::vec3);
+	uint32_t           index_buffer_size  = index_count * sizeof(uint32_t);
+	cube.index_count                      = index_count;
+
+	/* Array with vertices indexes for corresponding triangles */
+	std::array<uint32_t, index_count> indices{0, 4, 3, 7,
+	                                          UINT32_MAX,
+	                                          1, 0, 2, 3,
+	                                          UINT32_MAX,
+	                                          2, 6, 1, 5,
+	                                          UINT32_MAX,
+	                                          1, 5, 0, 4,
+	                                          UINT32_MAX,
+	                                          4, 5, 7, 6,
+	                                          UINT32_MAX,
+	                                          2, 3, 6, 7};
+
+	struct
+	{
+		VkBuffer       buffer;
+		VkDeviceMemory memory;
+	} vertex_pos_staging, vertex_norm_staging, index_staging;
```

Comment:
these are being used without being initialized.

Reply:
added initialization of members with VK_NULL_HANDLE